rust-runtime/aws-smithy-http-server/src/protocol/mod.rs

```diff
@@ -107,7 +107,7 @@ fn content_type_header_classifier(
             .parse::<mime::Mime>()
             // `expected_content_type` comes from the codegen.
             .expect("BUG: MIME parsing failed, `expected_content_type` is not valid. Please file a bug report under https://github.com/smithy-lang/smithy-rs/issues");
-        if expected_content_type != found_mime {
+        if expected_content_type != found_mime && expected_content_type != found_mime.essence_str() {
```

Comment:
Nice, I didn't know about `essence_str()`. Can't we simplify this to just the latter condition though?

```suggestion
        if expected_content_type != found_mime.essence_str() {
```

We control `expected_content_type` and it's a `&str` with just the essence string.  So if the latter pair are not equal the first pair should also not be equal.

Reply:
I did that originally, but was wondering if someone would ever want to force a specific check. It seemed unlikely, but added the original check back in, just in case. Given the discussion on it today, I think just the essence_str check is enough.